geniza/corpus/management/commands/merge_joins.py

```diff
@@ -187,27 +210,27 @@ def group_merge_candidates(self, joins):
                 report_rows.append(
                     [
                         shelfmark_type,
+                        group_id,
                         action,
                         status,
                         doc_status,
                         doc.id,
                         doc.description,
                     ]
                 )
+            group_id += 1
 
         self.stdout.write("%d groups with the same description" % same_desc)
+        return report_rows
 
-        if self.mode == "report":
-            self.generate_report(report_rows)
-
-    def generate_report(self, report_rows):
-        # output report of what would be done when in report mode
-
-        with open("merge-report.csv", "w") as csvfile:
+    def generate_report(self, report_rows, path):
+        """Generate a .csv report of identified candidate groups for merging."""
+        with open(path, "w") as csvfile:
```

Comment:
should add error handling here — raise a `CommandError` if the csv file is not found